### ISSUE TITLE
Services not loading/white screem

### ISSUE THREAD

**Issue by jancoow** [meetfranz/franz#253]
Hi,

I'm having an issue already a few months. Everytime a service reloads or redirect you to another page (like, if I want to login in skype and it moves me from the "fill in username page" to the "fill in password page") I get a white screen. So I'm unable to login into a few services. If the service doesn't reload or re-render the page, I can login or use it fine. 

### Expected Behavior
It should load/show/render the next page

### Current Behavior
I get a white screen instead

### Screenshots (if appropriate):

![image](https://user-images.githubusercontent.com/9000749/32629247-d25739fa-c598-11e7-8220-83c0439b60a7.png)

### Your Environment
Arch linux. 5.0.0_beta.13-1
But i'm having this issue already a few versions

**Comment by adlk** [comment]
This happens on every redirect?

**Comment by bobwinibago** [comment]
I have the same problem, every redirect only displays a white page.
But I can notice there is something loaded, because the cursor change depending on where it is on the page.
I had the same problem with Franz 4.
(I'm using Ubuntu 16.04 64bits)

**Comment by jancoow** [comment]
@adlk yes this happens on every redirect. Like @bobwinibago said, the page is loaded on the background. For example, when I login into steam, the page turns white. However, after restarting franz i'm logged in. In the old version the "reload page" button also worked. But this button is gone unfortunately.

**Comment by lordblendi** [comment]
I have the same problem, but for me restarting doesn't help. I want to log in on microsoft teams, it would redirect to my organization's page, but it turns white. Restarting it just brings me back to the login page

**Comment by adlk** [comment]
@jancoow to reload the service, you can use the menu (display it by pressing <kbd>Alt</kbd>) or directly use the reload shortcut <kbd>Ctrl</kbd>+<kbd>R</kbd>

@jancoow @bobwinibago @lordblendi Ad the white page issue: 
- Please open the service dev tools by pressing <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>Shift</kbd>+<kbd>i</kbd>
- Go to the Network tab
- Press F5
- Check every request that does not has a status 200. I'm especially interested in connections with a red status.

Thank you

**Comment by lordblendi** [comment]
@adlk I only have this red one
![image](https://user-images.githubusercontent.com/1140416/32662070-8d3d62cc-c629-11e7-97ba-34f70b3bd496.png)

**Comment by adlk** [comment]
That's good enough. You don't need to drop names but you are working in a big corporate environment? 

When connecting the dots of some error reports, I believe that SSO services and other 3rd party corporate-y security mechanism block requests. It's almost impossible for me to debug such things as I don't have the infrastructure ready but I can try to simulate it to a certain point.

It would be great if someone with a deeper dev knowledge who also happens to have access to such infrastructure is willing to help me on this one.

**Comment by bobwinibago** [comment]
Checked it with all 4 bugged services. Didn't found anything in common between them.
Here are the screenshots https://ufile.io/ehyen

**Comment by jancoow** [comment]
Sorry I was wrong. Reloading the service also makes the page white. So actually, a service is only loading on startup of the application. After I reload it or when I'm redirected the service turns white. I see a lot of errors in the console, which are some rendering errors. 

![image](https://user-images.githubusercontent.com/9000749/32669476-fafbd4a8-c640-11e7-9e3e-19312b8cfac3.png)

This error is spawned ~10 times per second. This maybe explains the ~15% cpu usage all the time.

**Comment by adlk** [comment]
@jancoow what services do you use? Do you have any custom services in your recipe folder?

**Comment by jancoow** [comment]
@adlk whatsapp, telegram, skype, steam, hangouts and messenger. No, I don't have any custom services.

**Comment by adlk** [comment]
@jancoow the steam bug is now fixed. Can you please open the Franz dev tools and check what error is displayed there that prevents you to run the app? <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>i</kbd>

**Comment by jancoow** [comment]
@adlk don't get me wrong. I have this issue with all of the services. It doesn't matter what service, if any of the services is redirecting me to another page it's white screen. Not only steam. I posted the console error a few comments back.

**Comment by adlk** [comment]
@jancoow you have reported several issues in here. Please just read the comments again, there is a difference between Service Dev Tools and Franz Dev Tools.

**Comment by jancoow** [comment]
@adlk Sorry, I was confused between these tools. I thought you was talking about the same dev tool. I can't open the franz dev tool with <kbd>ctrl</kbd> + <kbd>alt</kbd> + <kbd>i</kbd>. However, I can toggle it with  <kbd>ctrl</kbd> + <kbd>shift</kbd> + <kbd>i</kbd>, and there is some white box appearing on the right side of the window. But I guess this is also not rendered. It stays white. I think both the "service not rendering" and "Franz not rendering" after reload are somehow linked to each other.

**Comment by adlk** [comment]
Huh... in the case that even the dev tools don't work then I believe that something went terrible wrong when you initially built the app for Arch. I would suggest to remove everything and start from the beginning.

**Comment by lordblendi** [comment]
>You don't need to drop names but you are working in a big corporate environment?
Not that big, but yes, I do. And I don't need to be in company network to be able to connect.

**Comment by jancoow** [comment]
@adlk yes on one of my arch machines there is nothing wrong. But this one keeps having rendering issues in Franz. Maybe one of the dependencies is out dated.

**Comment by dvriv** [comment]
Hi, I am having the exact same issues than @jancoow with Arch Linux. Tried using the .Appimage and the AUR package without any success

**Comment by GHostlyFOX** [comment]
Ubuntu 17.10, Franz 5
similar problem

**Comment by MegaV0lt** [comment]
Same here. After reload the window stays empty:
![auswahl_20171121_09 15 41](https://user-images.githubusercontent.com/2804301/33062242-f0f7ef3a-ce9e-11e7-9adc-9c3ba2c4aaac.png)

Console after F5:
![auswahl_20171121_09 36 48](https://user-images.githubusercontent.com/2804301/33062424-8ffa385e-ce9f-11e7-8637-7d800842f00f.png)

**Comment by shinbyh** [comment]
I get the same issue only for the Google Hangouts.
My environment: Ubuntu 16.04 (amd64), franz version 5.0.0-beta13, loading two Slack messengers and one Skype.
The Hangout page works normally until login. Here I attach screenshots:
![franz_hangouts2](https://user-images.githubusercontent.com/15206035/33201225-f62e052c-d13a-11e7-8ae9-7434ae6fa1d0.png)
![franz_hangout_devtools](https://user-images.githubusercontent.com/15206035/33201231-f7f8513c-d13a-11e7-9dc8-05e8c12cbf3b.png)

**Comment by myroslav** [comment]
It should be some rendering bug. Since after I have "white window" bug, I locate a place where my cursor is "arrow", click there and drag significant distance, the service part of a window start displaing translucent overlay of service window being dragged around, and I see (but am hard to read due to poor contrast and translucency) proper content.

Thus I'm concluding that service works properly, and it is just display/rendering issue. Restart sometimes solves it for active service, but there is no guarantee it will for others.

**Comment by liefswanson** [comment]
having the same issue. Happens on most redirects, regardless of service. Skype is a particularly bad problem because it seems to redirect during load.

Definitely a rendering issue. I was able to interact with skype (put my password into the white screen). However, even after restarting the app, because skype redirects after login, it causes the screen to go white again.

![error](https://user-images.githubusercontent.com/5111290/33826284-568c96aa-de19-11e7-99a7-c59275a9f7b5.png)

This is the error skype gives me upon login, it does successfully log in though, I can hover over elements in the view with my mouse and they will display tooltips.

![tooltip](https://user-images.githubusercontent.com/5111290/33826432-d7e89802-de19-11e7-9bf2-ed03c4d5ecc4.png)

Running on Arch Linux using KDE with Linux 4.9lts.

**Comment by weierophinney** [comment]
I can verify as well, specifically with Skype. I'm running 5.0.0-beta.14, Release 1.7.9 / linux /x64 on Ubuntu 17.10, using X windows (vs wayland). 

Interestingly, it renders fine on my laptop, which is running the same Franz version and Ubuntu version, but not on my workstation.

**Comment by willcrobinson** [comment]
Same issue with all services when reloaded. Gnome Ubuntu 17.04 64-bit.

**Comment by cablop** [comment]
Same here. I thought it was a video issue... but now you are telling it fails to you too with Linux...
Failing here with Ubuntu 16.04 and 5.0.0-beta13 and 5.0.0-beta14.

**Comment by cablop** [comment]
After a few testing (i.e., trying to open the Service Dev Tools) i think the problem is a video issue... because the service tools also load in a white screen.

@weierophinney , is your workstation running an Open Source driver, an ATI video card, an integrated video card? Is your laptop using a proprietary driver, maybe nVidia or Intel?

**Comment by rokups** [comment]
Service tools work fine for me, not the service itself. Using AMDGPU driver here, RX580 gpu, mesa 17.2.6, archlinux.

**Comment by weierophinney** [comment]
> is your workstation running an Open Source driver, an ATI video card, an integrated video card? Is your laptop using a proprietary driver, maybe nVidia or Intel?

I have an ATI Radeon 6570, and am using the open-source driver (not the proprietary fglrx/catalyst one).

**Comment by willcrobinson** [comment]
I've got an AMD graphics card too. I do think that it's a video issue as the page acts as if it's loaded (and the pointer changes when I mouse over where the links would be in the page)

Will play about with my graphics drivers now to see if that changes anything.

**Comment by willcrobinson** [comment]
Definitely a graphics issue - installing the GPUPRO drivers from [here](http://support.amd.com/en-us/kb-articles/Pages/AMDGPU-PRO-Install.aspx) fixed the issue for me.

**Comment by dvriv** [comment]
RX480 too

**Comment by maxorlovsky** [comment]
Had same issue even in version 4, but mostly for Skype. In version 5+ it's happening for all apps.
Using latest ubuntu with franz 5.0.0-beta14, have integrated video card with intel processor.

This bug basically makes Franz unusable on Ubuntu, while on Windows everything works fine for me.

**Comment by weierophinney** [comment]
Confirmed as a graphics issue for me as well. I had some lingering fglrx drivers locally; once those were purged, and I was to stock open source radeon/amdgpu drivers, everything worked perfectly.

**Comment by bigzoo** [comment]
Just did fresh install (5.0.0-beta.14, Ubuntu 16) and all services are blank white. I filled the form ('invisibly') on a couple services and restarted and the work good.
Tried to sign in to keep and got a white screen again. Checked all request and all are 200 so not a network issue.
Got this interesting error though:
`
plugin.js:20 TypeError: require(...) is not a function
    at EventEmitter._electron.ipcRenderer.on (plugin.js:18)
    at emitTwo (events.js:106)
    at EventEmitter.emit (events.js:194)
`
plugin.js 17 & 18
`
// eslint-disable-next-line
    require(modulePath)(new RecipeWebview(), data);
`
This is a pretty common one which usually means that module.exports in `modulePath` hasn't been assigned to a function and hence making it impossible to call it as a function.

Gotta stop here for now, duty calls. Hope this sheds some light.

**Comment by berglh** [comment]
Ubuntu 17.10 here using version 5.0.0-beta.15 (on **Wayland**). I've been observing all the mentioned behaviours here.

Even when opening the Dev Tools <kbd>ctrl</kbd> + <kbd>shift</kbd> + <kbd>i</kbd> I get the white panel problem. i.e:

![franz](https://user-images.githubusercontent.com/12455338/35835503-d6965568-0b26-11e8-8af0-7da214e1ff8f.png)

<kbd>ctrl</kbd> + <kbd>shift</kbd> + <kbd>r</kbd> gives me the blue splash screen but no Franz intro logo.

I had the white service screen appearing mostly after adding a new service and attempting to auth. I often get the white screen on a service when I start Franz.

If I keep closing and re-opening Franz enough times, it eventually come good on all of my services. Seems transient in nature but happens enough to be annoying!

**Comment by antoinebelmont** [comment]
Any update here?
 i'm getting the same error, even skype in which i'm already logged; besides, when i run the app on command line, i get this error:
Gtk-Message: Failed to load module "pantheon-filechooser-module"

i don't know if that could help to track the issue.

Regards.

**Comment by bachrc** [comment]
Same here, and the AMDGPU PRO aren't working for my AMD. Can't use the app, too bad. :/ Is Electron in cause ?

**Comment by mimoralea** [comment]
It might not be the same issue, but I can tell you I had exactly the same issue and I'm ashamed to say it was that I was writing the full slack address "udacity-flying-car.slack.com" instead of "udacity-flying-car". So, make sure you are trying to reach the right service.

**Comment by rjconrad** [comment]
Wanted to switch to Franz from Rambox today, but this issue is a blocker for me. I'm running Ubuntu 17.10 with a Radeon graphics card using the default open source driver. When I added Slack and Whatsapp, everything was fine. Then I added Gmail and I got the white screen. When I restarted Franz, I could see the Gmail log in screen. Once I logged in, it was a white screen again. When I restarted Franz again, I could see my inbox. I also need Flowdock, which I need to log into via Google, but I never succeeded in seeing more than the first page.

**Comment by ArDrift** [comment]
I'm experiencing the same issue on Ubuntu 17.10, with a Radeon R7 240 graphics card, without amdgpu-pro drivers, as Ubuntu 17.10 is not supported. When I logged in the first time, I got a blank page, I had to kill the franz process as I enabled running in the background, and then I could see the pages correctly when I relaunched the app. If I want to log into another service, the issue is the same again, so it's something with the aforementioned redirects.

**Comment by Amandaville** [comment]
Exact same problems here.  franz-5.0.0-beta.16-x86_64.AppImage running on Linux Mint 18.1 with Radeon HD 6400M/7400M. Skype is a blank screen. Facebook messenger works unless I refresh.  Some menu operations cause the entire franz application to go blank and unresponsive. Clicking Help>About shows the message box with the franz version number, but closing the About message box results in the entire franz application turning to a blank blue screen.

**Comment by Amandaville** [comment]
It just occurred to me that although I have a Radeon card in my laptop, it's not the active card.  I'm only using the Intel integrated graphics. This thread seems to indicate that the problem is related to the Radeon card, so I wonder why am I experiencing the same symptoms with the integrated graphics.

**Comment by rokups** [comment]
Whatever cause for this issue is wavebox (electron app) works just fine with services which fail with franz, so this can be fixed.

**Comment by edmondtho** [comment]
I am also having 2 separate but similar issues:

1. GCal loads blank page initially - navigating around will populate the calendar. Navigating means:
-- selecting a day, close event creation dialogue box
-- navigate between months
-- hide/show calendars

2. Links open a blank window - Most (seems like all) links opened by any service results in a blank page in addition to the page being loaded
-- At most 1 blank page.

Here's a screenshot with both:
![screenshot from 2018-05-23 14-44-16](https://user-images.githubusercontent.com/39571782/40452899-45195e08-5e98-11e8-8d18-fb2bea0b9560.png)

OS: Ubuntu 16.04.4, was also occurring in Windows 10
Vid Card: Nvidia GeForce GT 640M
Franz Version: 5.0.0-beta.18

**Comment by dbogdanov** [comment]
Also having this issue with GCal.

**Comment by dbogdanov** [comment]
Any update on this?

**Comment by easyweb-su** [comment]
Have same issue in Telegram
![image](https://user-images.githubusercontent.com/10223187/49921469-a6c57580-febe-11e8-9a7f-95c45b5d5df2.png)
